code/modules/cm_aliens/weeds.dm

```diff
@@ -139,6 +139,21 @@
 	if(!W || !user || isnull(W) || (W.flags_item & NOBLUDGEON))
 		return 0
 
+	if(istype(W, /obj/item/sampler))
+		var/obj/item/sampler/samp = W
+		if(samp.filled)
+			return TRUE
+		health--
+		healthcheck()
+		samp.filled = TRUE
+		if(istype(src, /obj/effect/alien/weeds/node))
+			samp.sample = new /obj/item/marineResearch/xenomorph/weed/sack()
+			samp.update_icon()
+			return TRUE
+		samp.sample = new /obj/item/marineResearch/xenomorph/weed()
```

Comment:
Put this under an else to make the logic more clear. Leave a single `samp.update_icon()` and `return TRUE`.